cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/DynamicPartitioner.java

```diff
@@ -36,14 +36,14 @@
      * Will only allow writing to a new partition. If an attempt is made to write to a previously-existing partition,
      * a {@link DataSetException} will be thrown.
      */
-    NEW_ONLY,
+    CREATE,
     /**
      * Allows appending to an existing partition. If the partition does not already exist, it will be created.
      */
     ALLOW_APPEND,
     /**
-     * Overwrites the existing contents of the partition being written to. If the partition does not already exist, it
-     * will be created.
+     * Overwrites the existing partition being written to. It's contents as well as any related metadata will be
+     * overwritten. If the partition does not already exist, it will be created.
      */
     OVERWRITE
```

Comment:
but for consistency, either name them APPEND and OVERWRITE, or name them ALLOW_APPEND and ALLOW_OVERWRITE. Because both will create, but allow appending/overwriting if it already exists.